force-app/main/default/classes/RD2_OpportunityEvaluationService.cls

```diff
@@ -486,6 +488,28 @@ public inherited sharing class RD2_OpportunityEvaluationService {
         changeLogService.insertChangeLogRecords();
     }
 
+    private void evaluateSustainers() {
+        Set<Id> contactIds = UTIL_SObject.extractId(rds, npe03__Recurring_Donation__c.npe03__Contact__c);
+        Set<Id> accountIds = UTIL_SObject.extractId(rds, npe03__Recurring_Donation__c.npe03__Organization__c);
+
+        if (oldRds != null) {
+            contactIds.addAll(UTIL_SObject.extractId(oldRds, npe03__Recurring_Donation__c.npe03__Contact__c));
+            accountIds.addAll(UTIL_SObject.extractId(oldRds, npe03__Recurring_Donation__c.npe03__Organization__c));
+        }
+
+        RD2_SustainerEvaluationService sustainerService = new RD2_SustainerEvaluationService();
+        List<Contact> contactsToUpdate = sustainerService.evaulateSustainersFor('Contact', contactIds);
+        List<Account> accountsToUpdate = sustainerService.evaulateSustainersFor('Account', accountIds);
+
+        if (isBatchContext) {
```

Comment:
We only disable all triggers in the RD2 Nightly batch job and at the end of the transaction.